extensions/amp-ima-video/0.1/amp-ima-video.js

```diff
@@ -64,16 +70,34 @@ class AmpImaVideo extends AMP.BaseElement {
     assertHttpsUrl(this.element.getAttribute('data-tag'),
         'The data-tag attribute is required for <amp-video-ima> and must be ' +
             'https');
-    assertHttpsUrl(this.element.getAttribute('data-src'),
-        'The data-src attribute is required for <amp-video-ima> and must be ' +
-            'https');
+
+    // Set data-sources attribute based on source child elements.
+    const sourceElements = this.element.getElementsByTagName('source');
+    if (sourceElements.length > 0) {
+      const sources = [];
+      toArray(sourceElements).forEach(source => {
+        if (!this.preconnectSource_) {
+          this.preconnectSource_ = source.src;
+        }
+        const tmp = document.createElement('div');
+        tmp.appendChild(source);
+        sources.push(tmp./*REVIEW*/innerHTML);
```

Comment:
`tmp./OK/outerHTML` should be enough, no need for temp div. `outerHTML` is supported in browsers we care about and is becoming part of the spec.